lib/auth/sessions.go

```diff
@@ -363,6 +363,17 @@ func (a *Server) newWebSession(
 	if err != nil {
 		return nil, nil, trace.Wrap(err)
 	}
+
+	if tdr, err := a.calculateTrustedDeviceMode(ctx, func() ([]types.Role, error) {
```

Comment:
If we are doing this on newWebSession then could we remove the other invocations in favor of this one?

Reply:
Yes! good catch